llvm/lib/Analysis/MemoryDependenceAnalysis.cpp

```diff
@@ -1103,7 +1104,8 @@ bool MemoryDependenceResults::getNonLocalPointerDepFromBB(
         // be conservative.
         ThrowOutEverything =
             CacheInfo->Size.isPrecise() != Loc.Size.isPrecise() ||
-            CacheInfo->Size.getValue() < Loc.Size.getValue();
+            TypeSize::isKnownLT(CacheInfo->Size.getValue(),
```

Comment:
```suggestion
            !TypeSize::isKnownGE(CacheInfo->Size.getValue(),
```
I believe you will miss some invalidations otherwise.

Reply:
ok